src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -81,83 +70,79 @@ public ISecretManager GetSecretManager(WebHostSettings settings)
+                            // DateTime caches timezone information, so we need to clear the cache.
+                            TimeZoneInfo.ClearCachedData();
+                        }
 
-                    if (_standbyHostManager != null)
-                    {
-                        _standbyHostManager.Stop();
-                        _standbyHostManager.Dispose();
+                        if (_standbyHostManager != null)
+                        {
+                            _standbyHostManager.Stop();
+                            _standbyHostManager.Dispose();
+                        }
+                        _standbyReceiverManager?.Dispose();
+                        _standbyScriptHostConfig = null;
+                        _standbyHostManager = null;
+                        _standbyReceiverManager = null;
                     }
-                    _standbyReceiverManager?.Dispose();
-                    _standbyScriptHostConfig = null;
-                    _standbyHostManager = null;
-                    _standbyReceiverManager = null;
                 }
-            }
-            else
-            {
-                if (_standbyHostManager == null)
+                else
                 {
-                    var standbySettings = CreateStandbySettings(settings);
-                    _standbyScriptHostConfig = CreateScriptHostConfiguration(standbySettings, true);
-                    _standbyHostManager = new WebScriptHostManager(_standbyScriptHostConfig, _secretManagerFactory, _eventManager, _settingsManager, standbySettings);
-                    _standbyReceiverManager = new WebHookReceiverManager(_standbyHostManager.SecretManager);
-
-                    InitializeFileSystem();
-                    StandbyManager.Initialize(_standbyScriptHostConfig);
+                    // we're in standby (placeholder) mode
+                    if (_standbyHostManager == null)
+                    {
+                        var standbySettings = CreateStandbySettings(settings);
+                        _standbyScriptHostConfig = CreateScriptHostConfiguration(standbySettings, true);
+                        _standbyHostManager = new WebScriptHostManager(_standbyScriptHostConfig, _secretManagerFactory, _eventManager, _settingsManager, standbySettings);
+                        _standbyReceiverManager = new WebHookReceiverManager(_standbyHostManager.SecretManager);
+
+                        InitializeFileSystem();
+                        StandbyManager.Initialize(_standbyScriptHostConfig);
+
+                        // start a background timer to identify when specialization happens
+                        // specialization usually happens via an http request (e.g. scale controller
+                        // ping) but this timer is started as well to handle cases where we
+                        // might not receive a request
+                        _specializationTimer = new Timer(OnSpecializationTimerTick, settings, 1000, 1000);
```

Comment:
to avoid concurrent timer fired, ..., 1000, Timeout.Infinite).   Then, do a set when EnsureInitialize is done.

Reply:
I don't think there is an issue with concurrency here, so I'd prefer to keep the timer management as simple as possible.

E.g. we only start the time once we're successfully in standby mode and have the standby manager initialized. Subsequent calls to EnsureInitialized will then be noops, returning immediately. When we're specializing, we stop the timer immediately before setting the active manager.